app/Core/Ldap/User.php

```diff
@@ -88,16 +88,34 @@ public function find($query)
      * @param  string  $username
      * @return string[]
      */
+    protected function getGroupsRecursive(Entry $entry, $member, &$groupsAll)
+    {
+
+        $groups = $this->group->find(sprintf($this->getGroupUserFilter(), $member));
+
+        foreach ($groups as $group) {
+            $groupId = $group->getExternalId();
+
+            if (!isset($groupsAll[$groupId])) {
+                $groupsAll[$groupId] = $member;
+                $this->getGroupsRecursive($entry, $groupId, $groupsAll);
+            }
+        }
+    }
+
     protected function getGroups(Entry $entry, $username)
     {
         $groupIds = array();
 
+        // If LDAP_GROUP_FILTER has been filled, use it to get groups referring to user
         if (! empty($username) && $this->group !== null && $this->hasGroupUserFilter()) {
-            $groups = $this->group->find(sprintf($this->getGroupUserFilter(), $username));
+            // Try with the dn (eg: member=uid=username,ou=accounts,dc=local)
+            $this->getGroupsRecursive($entry, $entry->getDn(), $groupIds);
+            // try with the username (eg: memberUid=username)
+            $this->getGroupsRecursive($entry, $username, $groupIds);
```

Comment:
Why do you need to try with the DN and then username?

Reply:
Some implementations are done with a weak reference to the name, like (posixGroup + memberuid = username), while some other are done with the strong reference of the DN (groupOfNames + member = uid=username,ou=people,dc=local).
This strong reference is sometimes prefered, as the overlay can update all references when the targeted object is renamed.

Before, it was only done with the username, thus only working with weak references. Now I try with both, as it's easier than letting the user choose in the config, and introducing a new "replacement format", like "%user% or "%dn%".